src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -100,17 +108,80 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 while (userName.Length == 0);
             }
 
-            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName
-            );
+            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName);
 
             //
             // now, prompt for the password
             //
-            WriteToConsole(passwordPrompt, true);
-            password = ReadLineAsSecureString();
-            if (password == null)
+            if (confirmPassword)
             {
-                return null;
+                confirmPasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_ConfirmPassword, userName);
+                bool passwordsMatch;
+                do
+                {
+                    WriteToConsole(passwordPrompt, true);
+                    password = ReadLineAsSecureString();
+                    WriteToConsole(confirmPasswordPrompt, true);
+                    confirmedPassword = ReadLineAsSecureString();
+                    
+                    IntPtr passwordBstr = IntPtr.Zero;
```

Comment:
Please move the comparison code to private temporary method.

Reply:
I moved it to a private method in the class, not entirely sure what you're referring to around it being temporary. If the new private method isn't acceptable just let me know what we'd need to change.

Reply:
Private method is that I meant. It simplify the code and allow us to replace comparison algorithm later if needed.